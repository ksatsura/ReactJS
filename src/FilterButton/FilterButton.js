//imports
import React from 'react';
import '../style.css';

const filterButton = (props) => {
    return (
        <button id={props.id} className={props.styles} onClick={props.onClick}>{props.id}</button>
    );
};

export default filterButton;