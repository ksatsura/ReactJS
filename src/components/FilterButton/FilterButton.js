//imports
import * as React from 'react';
import classNames from 'classnames';
import { connect } from  'react-redux';
import { toggleSearchFilter, toggleSortFilter } from '../../actions/searchActions';
import '../../style.css';

// @flow
type Props = {
    id: string, 
    searchBy: string, 
    sortBy: string, 
    handleFilterClick: (id: string) =>  void, 
    handleSortClick: (id: string) =>  void, 
    name: string,
};

export const FilterButton = (props: Props) => {

    const { id, searchBy, sortBy, handleFilterClick, handleSortClick, name } = props;
    const handleClick = (id === 'title' || id === 'genre') ? handleFilterClick : handleSortClick;
    let styles = (id === 'title' || id === 'genre') 
        ? classNames({ [`search-filter-btn-${id}`]: true }, { focused: (searchBy === id) })
        : classNames('search-results-btn', { selected: (sortBy === id) });

    return (
        <button id={id} className={styles} onClick={(e) => handleClick(e.target.id) }>{name}</button>
    );
    
};

const mapStateToProps = state => {
    return {
        searchBy: state.filmListReducer.searchBy,
        sortBy: state.filmListReducer.sortBy,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        handleFilterClick: (id) => dispatch(toggleSearchFilter(id)),
        handleSortClick: (id) => dispatch(toggleSortFilter(id)),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilterButton);