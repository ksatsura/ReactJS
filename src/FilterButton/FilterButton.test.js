import React from 'react';
import FilterButton from './FilterButton.js';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';

describe('FilterButton component', () => {

    let styles;
    let handleClick;

    beforeAll(() => {
        styles='';
        handleClick=jest.fn();
    });

    it('should be render correctly', () => {
        const component = shallow(<FilterButton styles={styles} onClick={handleClick} />);
        
        expect(component).toMatchSnapshot();
    });

    it('test filter button click event', () => {
        
        const component = shallow(<FilterButton styles={styles} onClick={handleClick}/>);

        component.find('button').simulate('click');
        expect(handleClick.mock.calls.length).toEqual(1);
    });
});