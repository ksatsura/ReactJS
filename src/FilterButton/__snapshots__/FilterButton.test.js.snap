// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton component should be render correctly 1`] = `ShallowWrapper {}`;
